torch/distributed/fsdp/_common_utils.py

```diff
@@ -39,6 +41,54 @@
 FSDP_FLATTENED = "_fsdp_flattened"
 
 
+class _FSDPDeviceHandler:
+    """
+    This is a simple abstraction for FSDP computing devices,
+    which enables custom backends that implement CUDA-like
+    semantics to be integrated with FSDP.
+    """
+
+    def __init__(self, device: torch.device, backend: Any = None):
+        if backend is None:
+            try:
+                self.__backend = getattr(torch, device.type)
+                self.__device = device
+            except AttributeError:
+                raise AttributeError(
+                    f"Device '{device}' does not have a corresponding backend registered as 'torch.{device.type}'."
+                )
+        else:
+            self.__backend = backend
+
+    @classmethod
+    def from_device(cls, device: torch.device) -> "_FSDPDeviceHandler":
+        """
+        Return an device handle corresponding to the device, and through this handle,
```

Comment:
nit: Since you refer to the object as a "device handle", should we rename the class to `_FSDPDeviceHandle`? (If so, we should also rename the attribute saved to FSDP.)